javascript/src/router.js

```diff
@@ -36,6 +56,21 @@ function routeAppliesToCurrentLocation(route) {
   return r.match(page.current, {});
 }
 
+/**
+ * Find base url if available
+ *
+ * @returns {string}
+ */
+function getAbsoluteBase() {
+  const baseTags = window.document.getElementsByTagName('base');
+  if (baseTags && baseTags[0]) {
+    return baseTags[0].href;
+  }
+  return null;
+}
+
+page.getAbsoluteBase = getAbsoluteBase.bind(page);
+page.normalise = normalise.bind(page);
```

Comment:
Does this need to be added as a public API? If we add it, we're committing to support it.